activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -58,6 +58,18 @@ def schema_creation
         SchemaCreation.new self
       end
 
+      def column_spec_for_primary_key(column)
+        spec = {}
+        if column.extra == 'auto_increment'
+          return unless column.limit == 8
+          spec[:id] = ':bigint'
+        else
+          spec[:id] = column.type.inspect
+          spec.merge!(prepare_column_options(column).delete_if { |key, _| [:name, :type, :null].include?(key) })
```

Comment:
Code fails there as `prepare_column_options` in 4.2 requires two arguments: `column` and `types`. Second argument must be supplied (I do not know what is it).

ActiveRecord internals differs heavily between 4.2 and 5.0, so it wouldn't be easy to backport anything. But I need this fix too.